cmd/fetch.go

```diff
@@ -43,3 +47,21 @@ func Fetch(ctx *cli.Context) {
 
 	hw.Summarize()
 }
+
+func setSubmissionState(c *config.Config, problems []*api.Problem) error {
```

Comment:
This PR looks great! What do you think about passing the `map[string][]SubmissionInfo` in to this function ? I think that would make it easier to test.